app/scripts/controllers/transactions/index.js

```diff
@@ -146,6 +150,49 @@ export default class TransactionController extends EventEmitter {
     return integerChainId;
   }
 
+  /**
+   * @ethereumjs/tx uses @ethereumjs/common as a configuration tool for
+   * specifying which chain, network, hardfork and EIPs to support for
+   * a transaction. By referencing this configuration, and analyzing the fields
+   * specified in txParams, @ethereumjs/tx is able to determine which EIP-2718
+   * transaction type to use.
+   * @returns {Common} common configuration object
+   */
+  getCommonConfiguration() {
+    // type will be one of our default network names or 'rpc'. the default
+    // network names are sufficient configuration, simply pass the name as the
+    // chain argument in the constructor.
+    const { type, nickname } = this.getProviderConfig();
+
+    if (type === 'rpc') {
+      // For 'rpc' we need to use the same basic configuration as mainnet,
+      // since we only support EVM compatible chains, and then override the
+      // name, chainId and networkId properties. This is done using the
+      // `forCustomChain` static method on the Common class.
+      const chainId = this._getCurrentChainId();
+      const networkId = this.networkStore.getState();
+
+      const customChainParams = {
+        name: nickname,
+        chainId: parseInt(chainId, 16),
```

Comment:
Minor but can we use `parseInt` when we first create the variable, since it's not used as a string in this function?  Would be cool to know its type right away.  Not a blocker, just a suggestion.

Reply:
@darkwing done in latest. Also renamed nickname -> name during restructuring